packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -259,6 +259,25 @@ class RecordActionTool implements RecorderTool {
     });
   }
 
+  onContextMenu(event: MouseEvent) {
```

Comment:
Please add a comment that the 'contextmenu' event is triggered by a right-click or equivalent action, and it prevents the click event from firing for that action, so we always convert 'contextmenu' into a right-click.

Reply:
https://github.com/microsoft/playwright/pull/31634/commits/d9bc5fcfcce43db5870f907910f55f05c91399a4